### PR TITLE
Fix TODO's

### PR DESCRIPTION
**Pull request by markwallace-microsoft**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: samples/apps/github-qna-webapp-react/src/components/GitHubRepoSelection.tsx [hunk 1]
```diff
@@ -23,25 +23,29 @@ const GitHubProjectSelection: FC<IData> = ({ uri, keyConfig, onLoadProject, onBa
...
+                },
+                'GitHubSkill',
                 'SummarizeRepository',
             );
             setIsLoaded(true);
             console.log(result);
         } catch (e) {
             setIsLoadError(true);
-            alert('Something went wrong.\n\nDetails:\n' + e);
+            alert('Something went wrong. Please check that the function is running and accessible from this location.');
```
Comment (TaoChenOSU) [inline samples/apps/github-qna-webapp-react/src/components/GitHubRepoSelection.tsx]: I think you might need to rebase to get the latest changes.

File: samples/dotnet/github-skills/GitHubSkill.cs [hunk 2]
```diff
@@ -184,4 +156,56 @@ public async Task SummarizeRepositoryAsync(string source, SKContext context)
...
+    /// Summarize a code file into an embedding
+    /// </summary>
+    private async Task SummarizeCodeFileAsync(string filePath, string repositoryUri, string repositoryBranch, string fileUri)
+    {
+        string code = File.ReadAllText(filePath);
+
+        if (code != null && code.Length > 0)
+        {
+            string text;
+            if (code.Length < MaxFileSize)
```
Comment (glahaye) [inline samples/dotnet/github-skills/GitHubSkill.cs]: Change side for this inequality: '<' to '>'

**Review by glahaye — CHANGES_REQUESTED** [review]
Just that inequality sign.  After that, it will be ready for merge as far as I'm concerned

File: samples/dotnet/github-skills/GitHubSkill.cs [hunk 3]
```diff
@@ -149,27 +113,35 @@ public async Task SummarizeCodeDirectoryAsync(string directoryPath, SKContext co
...
+        if (!context.Variables.Get(Parameters.RepositoryBranch, out string repositoryBranch) || string.IsNullOrEmpty(repositoryBranch))
+        {
+            repositoryBranch = "main";
+        }
+        if (!context.Variables.Get(Parameters.SearchPattern, out string searchPattern) || string.IsNullOrEmpty(searchPattern))
+        {
+            searchPattern = "*.md";
+        }
+
         string filePath = Environment.ExpandEnvironmentVariables($"%temp%\\SK-{Guid.NewGuid()}.zip");
```
Comment (craigomatic) [inline samples/dotnet/github-skills/GitHubSkill.cs]: Given you are creating a file in the temp dir you could replace %temp%\etc with a Path.Combine(Path.GetTempPath(), etc)

File: samples/dotnet/github-skills/GitHubSkill.cs [hunk 4]
```diff
@@ -149,27 +113,35 @@ public async Task SummarizeCodeDirectoryAsync(string directoryPath, SKContext co
...
         string directoryPath = Environment.ExpandEnvironmentVariables($"%temp%\\SK-{Guid.NewGuid()}");
```
Comment (craigomatic) [inline samples/dotnet/github-skills/GitHubSkill.cs]: Ditto

**Pull request merged**